src/main/jbake/content/developer/jms/advanced.adoc

```diff
@@ -0,0 +1,112 @@
+= JMS Advanced
+:jbake-date: 2016-03-17
+:jbake-type: page
+:jbake-status: published
+:jbake-tomeepdf:
+
+
+=== Configuring the Container tomee.xml
+
+===== Queue
+
+[source,java]
+----
+<Resource id="Foo" type="javax.jms.Queue">
+    destination = Specifies the name of the queue
+</Resource>
+----
+
+===== Topic
+
+[source,java]
+----
+
+<Resource id="Foo" type="javax.jms.Topic">
+    destination = Specifies the name of the topic
+</Resource>
+----
+
+
+===== ConnectionFactory
+
+[source,java]
+----
+
+<Resource id="Foo" type="javax.jms.ConnectionFactory">
+    ResourceAdapter = Default value is Default JMS Resource Adapter.
+    TransactionSupport = Specifies if the connection is enrolled in global transaction, allowed values: xa, local or none. Default value is xa.
+    PoolMaxSize = Maximum number of physical connection to the ActiveMQ broker. Default value is 10.
+    PoolMinSize  = Minimum number of physical connection to the ActiveMQ broker. Default value is 0.
+    ConnectionMaxWaitMilliseconds = Maximum amount of time to wait for a connection. Default value is 5000.
+    ConnectionMaxIdleMinutes = Maximum amount of time a connection can be idle before being reclaimed. Default value is 15.
+</Resource>
+
+----
+
+===== MESSAGE
+
+[source,java]
+----
+
+<Container id="Foo" type="MESSAGE">
+    ResourceAdapter = The resource adapter delivers messages to the container. Default value is Default JMS Resource Adapter.
+    MessageListenerInterface = Specifies the message listener interface handled by this container. Default value is javax.jms.MessageListener.
+    ActivationSpecClass = Specifies the activation spec class. Default value is org.apache.activemq.ra.ActiveMQActivationSpec.
+    InstanceLimit = Specifies the maximum number of bean instances that are allowed to exist for each MDB deployment. Default value is 10.
+</Container>
+
+----
+
+
+=== Activation Message
+
+A JMS activation specification is associated with one or more message-driven beans and provides the configuration necessary for them to receive messages.
+
+
+[source,java]
+----
+containerId.activation.atrribute=value
+----
+
+So, the attribute will send through all the message-drive bean who has the container id equals to *containerId*.
+
+
+==== Ignore fields:
+
+To ignore any field from Activation message, override the *failOnUnknownActivationSpec* to _false_ at MdbContainer class.
+Add a file `service-jar.xml` at the path `resources/META-INF/org.apache.openejb/`
+
+[source,xml]
+----
+
+ <ServiceProvider
```

Comment:
You actually don't need to use a `service-jar.xml` - that's a little misleading. The `service-jar.xml` actually provides all the defaults for various resources - so if you use `@Resource` with a `ConnectionFactory` that has not been defined, the defaults in the `service-jar.xml` file are used.

Instead you can simply create a `Container` in `tomee.xml` and add the `FailOnUnknownActivationSpec = false` setting to it, like so:

```
  <Container id="MyMDBContainer" ctype="MESSAGE">
    ResourceAdapter=MyResourceAdapter
    ActivationSpecClass=my.JmsActivationSpec
    InstanceLimit=20
    FailOnUnknownActivationSpec = false
  </Container>
```

This _overrides_ the definition in `service-jar.xml`.

If that doesn't make sense, do let me know - I'll try and explain it better. Its a useful concept to know about.